### PR TITLE
Allow the user to specify tls InsecureSkipVerify and ServerName

### PR DESCRIPTION
**Pull request by jmcfarlane**

### CODE REVIEW

File: ldap-client.go [hunk 1]
```diff
@@ -41,7 +45,10 @@ func (lc *LDAPClient) Connect() error {
 				return err
 			}
 		} else {
-			l, err = ldap.DialTLS("tcp", address, &tls.Config{InsecureSkipVerify: false})
+			l, err = ldap.DialTLS("tcp", address, &tls.Config{
+				InsecureSkipVerify: lc.InsecureSkipVerify,
```
Comment (jtblin) [inline ldap-client.go]: For my understanding, why doing that instead of passing `UseSSL` to `true`? 
And what does passing `ServerName` change?

Comment (jmcfarlane) [inline ldap-client.go]: When `UseSSL` is true, you **must** either specify `ServerName` or enable `InsecureSkipVerify`.

Reference: https://golang.org/pkg/crypto/tls/#Client

Comment (jmcfarlane) [inline ldap-client.go]: Yep, makes sense :)

This commit effectively provides a mechanism for the user to enable ssl. If they want to keep cert verification they can specify `ServerName` and if they don't, they can disable verification. I suppose this project could set the default `ServerName` based on `Host` while keeping cert verification. The current approach honors the golang montra of "do less enable more" :)

Comment (jmcfarlane) [inline ldap-client.go]: Any thoughts on if this could be merged?

Comment (jmcfarlane) [inline ldap-client.go]: For additional detail, this is the upstream commit that requires this change:

https://github.com/golang/go/commit/fca335e91a915b6aae536936a7694c4a2a007a60

Comment (jtblin) [inline ldap-client.go]: OK I did some more testing and it's the only way to make it work with `useSSL=true`. Thanks a ton for your patience and for the PR!

**Pull request merged**